cf-testd/cf-testd.c

```diff
@@ -471,12 +542,49 @@ int main(int argc, char *argv[])
     if (config->file != NULL)
     {
         Log(LOG_LEVEL_NOTICE, "Got file argument: '%s'", config->file);
-        Log(LOG_LEVEL_ERR, "File input not yet supported");
-        // TODO: implement file input
-        exit(EXIT_FAILURE);
+        FILE *fp = NULL;
+        char buffer[CF_BUFSIZE];
+        int c, i = 0;
+
+        if ((fp = fopen(config->file, "r")) == NULL)
```

Comment:
Don't fopen inside condition. Move it up like my comment above suggests, and just do:
```C